cosmos/cache.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import shutil
+from enum import Enum
+from pathlib import Path
+
+from cosmos import settings
+from cosmos.constants import DBT_PARTIAL_PARSE_FILE_NAME, DBT_TARGET_DIR_NAME
+from cosmos.dbt.project import get_partial_parse_path
+
+
+def create_cache_identifier(dag_name: str, task_group_name: str = "") -> str:
```

Comment:
Need on the usage this should be

```suggestion
def create_cache_identifier(dag_name: str, task_group_name: str | None = None) -> str:
```